pkg/util/k8sutil/pod_util.go

```diff
@@ -70,18 +70,22 @@ func containerWithRequirements(c v1.Container, r v1.ResourceRequirements) v1.Con
 
 func newEtcdProbe(isSecure, isTLSSecret bool) *v1.Probe {
 	// etcd pod is healthy only if it can participate in consensus
-	cmd := "ETCDCTL_API=3 etcdctl endpoint status"
+	var cmd []string
+	cmd = append(cmd, "etcdctl")
 	if isSecure {
-		tlsFlags := fmt.Sprintf("--cert=%[1]s/%[2]s --key=%[1]s/%[3]s --cacert=%[1]s/%[4]s", operatorEtcdTLSDir, etcdutil.CliCertFile, etcdutil.CliKeyFile, etcdutil.CliCAFile)
+		cmd = append(cmd, fmt.Sprintf("--endpoints=https://localhost:%d", EtcdClientPort))
 		if isTLSSecret {
```

Comment:
You could decide the file names and them build the command, so you don't have to repeat the parameter names.
```
if isTLSSecret {
    certFile = 'tls.crt'
    keyfile = 'tls.key',
    ...
}
else
{
    certFile = etcdutil.CliCertFile
    ....
}
 cmd.append(cmd, ....)
```

Reply:
That's a nice idea, thx